Traceback (most recent call last):
  File "D:\anaconda3\lib\site-packages\jupyter_cache\executors\utils.py", line 58, in single_nb_execution
    executenb(
  File "D:\anaconda3\lib\site-packages\nbclient\client.py", line 1204, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "D:\anaconda3\lib\site-packages\nbclient\util.py", line 84, in wrapped
    return just_run(coro(*args, **kwargs))
  File "D:\anaconda3\lib\site-packages\nbclient\util.py", line 62, in just_run
    return loop.run_until_complete(coro)
  File "D:\anaconda3\lib\asyncio\base_events.py", line 647, in run_until_complete
    return future.result()
  File "D:\anaconda3\lib\site-packages\nbclient\client.py", line 663, in async_execute
    await self.async_execute_cell(
  File "D:\anaconda3\lib\site-packages\nbclient\client.py", line 965, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "D:\anaconda3\lib\site-packages\nbclient\client.py", line 862, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
# Get data
data_behavior = get_data_simulation(n, seed, policy="behavior")
#data_target = get_data_simulation(n0, seed, policy="target")

# The true expected heterogeneous treatment effect
HTE_true = get_data_simulation(n, seed, policy="1")['R']-get_data_simulation(n, seed, policy="0")['R']


------------------

[1;31m---------------------------------------------------------------------------[0m
[1;31mUnboundLocalError[0m                         Traceback (most recent call last)
Input [1;32mIn [3][0m, in [0;36m<cell line: 6>[1;34m()[0m
[0;32m      2[0m data_behavior [38;5;241m=[39m get_data_simulation(n, seed, policy[38;5;241m=[39m[38;5;124m"[39m[38;5;124mbehavior[39m[38;5;124m"[39m)
[0;32m      3[0m [38;5;66;03m#data_target = get_data_simulation(n0, seed, policy="target")[39;00m
[0;32m      4[0m 
[0;32m      5[0m [38;5;66;03m# The true expected heterogeneous treatment effect[39;00m
[1;32m----> 6[0m HTE_true [38;5;241m=[39m [43mget_data_simulation[49m[43m([49m[43mn[49m[43m,[49m[43m [49m[43mseed[49m[43m,[49m[43m [49m[43mpolicy[49m[38;5;241;43m=[39;49m[38;5;124;43m"[39;49m[38;5;124;43m1[39;49m[38;5;124;43m"[39;49m[43m)[49m[[38;5;124m'[39m[38;5;124mR[39m[38;5;124m'[39m][38;5;241m-[39mget_data_simulation(n, seed, policy[38;5;241m=[39m[38;5;124m"[39m[38;5;124m0[39m[38;5;124m"[39m)[[38;5;124m'[39m[38;5;124mR[39m[38;5;124m'[39m]

File [1;32mD:\anaconda3\lib\site-packages\causaldm\_util_causaldm.py:230[0m, in [0;36mget_data_simulation[1;34m(n0, seed, policy)[0m
[0;32m    227[0m     data[38;5;241m=[39m{[38;5;124m'[39m[38;5;124mS1[39m[38;5;124m'[39m:S_1,[38;5;124m'[39m[38;5;124mS2[39m[38;5;124m'[39m:S_2,[38;5;124m'[39m[38;5;124mA[39m[38;5;124m'[39m:A,[38;5;124m'[39m[38;5;124mR[39m[38;5;124m'[39m:R}
[0;32m    228[0m     data[38;5;241m=[39mpd[38;5;241m.[39mDataFrame(data)
[1;32m--> 230[0m [38;5;28;01mreturn[39;00m [43mdata[49m

[1;31mUnboundLocalError[0m: local variable 'data' referenced before assignment
UnboundLocalError: local variable 'data' referenced before assignment

